packages/data-stores/src/queries/use-has-seen-whats-new-modal-query.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback } from 'react';
+import { useQuery, useQueryClient, useMutation } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+
+interface HasSeenWhatsNewModal {
+	hasSeenWhatsNewModal: boolean;
+}
+
+interface HasSeenWhatsNewModalResult {
+	has_seen_whats_new_modal: boolean;
+}
+
+interface UpdateError {
+	message: string;
+	error: string;
+}
+
+export const useHasSeenWhatsNewModalQuery = ( siteId: number | null ) => {
+	const queryKey = 'has-seen-whats-new-modal';
+
+	const { data, isLoading } = useQuery< { has_seen_whats_new_modal: boolean } >(
+		queryKey,
+		async () =>
+			await wpcomRequest( {
+				path: `/sites/${ siteId }/block-editor/has-seen-whats-new-modal`,
+				apiNamespace: 'wpcom/v2',
+			} ),
+		{
+			enabled: !! siteId,
+		}
+	);
+
+	const queryClient = useQueryClient();
+	const mutation = useMutation< HasSeenWhatsNewModalResult, UpdateError, HasSeenWhatsNewModal >(
+		async ( { hasSeenWhatsNewModal } ) =>
+			await wpcomRequest( {
+				path: `/sites/${ siteId }/block-editor/has-seen-whats-new-modal`,
+				apiNamespace: 'wpcom/v2',
+				method: 'post',
+				body: {
+					has_seen_whats_new_modal: hasSeenWhatsNewModal,
+				},
+			} ),
+		{
+			onSuccess( data ) {
+				queryClient.setQueryData( queryKey, {
+					...data,
+				} );
+			},
+		}
+	);
+
+	const { mutate } = mutation;
+
+	const setHasSeenWhatsNewModal = useCallback(
+		( hasSeenWhatsNewModal: boolean ) => {
+			mutate( { hasSeenWhatsNewModal } );
+		},
+		[ mutate ]
+	);
```

Comment:
I found out recently that there is also `mutateAsync`. I prefer it over this one because it returns a promise, which is useful to know when things are done.

In this case, we know we don't care about the results, but in the future, I'd return the promise from `mutateAsync` so the API users get more control out of the box.